libkbfs/disk_block_cache_test.go

```diff
@@ -447,12 +455,56 @@ func TestDiskBlockCacheDynamicLimit(t *testing.T) {
 		" evicted each time we went past the limit.")
 	start := numBlocks - int(defaultNumBlocksToEvict)
 	for i := 1; i <= numBlocks; i++ {
-		blockID, blockEncoded, serverHalf := setupBlockForDiskCache(t, config)
-		err := cache.Put(ctx, tlf.FakeID(10, false), blockID, blockEncoded, serverHalf)
+		blockPtr, _, blockEncoded, serverHalf := setupBlockForDiskCache(
+			t, config)
+		err := cache.Put(ctx, tlf.FakeID(10, false), blockPtr.ID, blockEncoded, serverHalf)
 		require.NoError(t, err)
 		require.Equal(t, start+(i%int(defaultNumBlocksToEvict)), cache.numBlocks)
 	}
 
 	require.True(t, int64(cache.currBytes) < currBytes)
 	require.Equal(t, start, cache.numBlocks)
 }
+
+func TestDiskBlockCacheWithRetrievalQueue(t *testing.T) {
+	t.Parallel()
+	t.Log("Test the interaction of the disk block cache and retrieval queue.")
+	dbc, dbcConfig := initDiskBlockCacheTest(t)
+	require.NotNil(t, dbc)
+	defer shutdownDiskBlockCacheTest(dbc)
+
+	t.Log("Create a queue with 0 workers to rule it out from serving blocks.")
+	bg := newFakeBlockGetter(false)
+	q := newBlockRetrievalQueue(0, newTestBlockRetrievalConfig(t, bg, dbc))
+	require.NotNil(t, q)
+	defer q.Shutdown()
+
+	ctx := context.Background()
+	kmd := makeKMD()
+	ptr1, block1, block1Encoded, serverHalf1 := setupBlockForDiskCache(
+		t, dbcConfig)
+	err := dbc.Put(ctx, kmd.TlfID(), ptr1.ID, block1Encoded, serverHalf1)
+	_, _ = bg.setBlockToReturn(ptr1, block1)
+	require.NoError(t, err)
```

Comment:
Would be nice to swap this with the line above so the err checking happens right away.